lib/src/components/NxCollapsibleItemsSelect/NxCollapsibleMultiSelect/__tests__/NxCollapsibleMultiSelect.test.tsx

```diff
@@ -4,355 +4,559 @@
-      isOpen: true,
-      disabled: true,
-      onToggleCollapse: jest.fn(),
-      disabledTooltip: 'test disabled tooltip',
-      optionTooltipGenerator: (option: Option) => option.name,
-      tooltipModifierClass: 'tooltip-test-class',
-      onFilterChange: jest.fn(),
-      filter: 'filter term',
-      filteredOptions: [{id: 'bar', name: 'Bar'}],
-      filterPlaceholder: 'test filter placeholder',
-      filterThreshold: 1,
-      id: 'someid'
-    };
-
-    const shallowRender = getShallow(optionalProps);
-    expect(shallowRender).toHaveDisplayName('AbstractCollapsibleItemsSelect');
-    expect(shallowRender).toHaveProp(optionalProps);
+  describe('exceptions', function() {
+    beforeEach(function() {
+      // prevent RTL logging thrown exceptions
+      jest.spyOn(console, 'error').mockImplementation(() => {});
+    });
+
+    it('throws an error if a selectedId is supplied that is not part of the options', function() {
+      const expectedErr = 'You are attempting to select "ufo", but it is not part of the available options';
+
+      expect(() => quickRender({ selectedIds: new Set() })).not.toThrow();
+      expect(() => {
+        quickRender({ selectedIds: new Set(['ufo']) });
+      }).toThrow(expectedErr);
+    });
   });
 
-  describe('renderOption prop', function () {
-    it('renders unchecked <NxCheckbox> if option is not selected', function () {
-      const shallowRender = getShallow({
-        selectedIds: new Set(['foo'])
-      });
-      const optionWrapper = shallowRender.renderProp('renderOption')({id: 'bar', name: 'Bar'});
+  describe('trigger', function() {
+    const renderAndGetTrigger = (props?: Partial<Props>) => quickRender(props).getByRole('button');
 
-      expect(optionWrapper).toHaveDisplayName('ForwardRef(NxCheckbox)');
-      expect(optionWrapper).toHaveProp('isChecked', false);
+    it('is a button', function() {
+      expect(renderAndGetTrigger()).toBeInTheDocument();
     });
 
-    it('renders checked <NxCheckbox> if option is selected', function () {
-      const shallowRender = getShallow({
-        selectedIds: new Set(['foo'])
-      });
-      const optionWrapper = shallowRender.renderProp('renderOption')({id: 'foo', name: 'Foo'});
+    it('has type="button"', function() {
+      expect(renderAndGetTrigger()).toHaveAttribute('type', 'button');
+    });
+
+    it('references the children items using aria-controls', function() {
+      const view = quickRender(),
+          trigger = view.getByRole('button'),
+          childrenElId = trigger.getAttribute('aria-controls')!;
+
+      expect(childrenElId).toBeDefined();
+
+      const childrenEl = document.getElementById(childrenElId)!,
+          childEl = view.getByRole('menu');
 
-      expect(optionWrapper).toHaveDisplayName('ForwardRef(NxCheckbox)');
-      expect(optionWrapper).toHaveProp('isChecked', true);
+      expect(view.container).toContainElement(childrenEl);
+      expect(childrenEl).toContainElement(childEl);
```

Comment:
I'd think we want to assert that the element referenced by `aria-controls` has the `menu` role itself, not just contains an element that has it.

Reply:
Updated here. b8784b45f5ff1e74bfa592c0845358924c2cb56e